sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_responses.py

```diff
@@ -0,0 +1,38 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+from azure.core.utils import CaseInsensitiveDict
+
+
+class CosmosDictResponse(dict):
+    def __init__(self, original_dict, response_headers):
+        if original_dict is None:
+            original_dict = {}
+        super().__init__(original_dict)
+        self._response_headers = response_headers
+
+    @property
+    def response_headers(self) -> CaseInsensitiveDict:
```

Comment:
I think we might need to treat this as a dict type?
```suggestion
    def response_headers(self) -> CaseInsensitiveDict[str, Any]:
```
Or maybe the headers are actually just `[str, str]`?

Reply:
not all headers we return are strings since some get set in the SDK as opposed to the service

Reply:
also, mypy complains about this:
![image](https://github.com/user-attachments/assets/344a0d24-8a3e-47c3-8bbb-72dbb3625a6b)

Reply:
It would be very unexpected to have HTTP headers with a non-string value returned...